django/db/backends/base/client.py

```diff
@@ -21,6 +21,5 @@ def settings_to_cmd_args_env(cls, settings_dict, parameters):
 
     def runshell(self, parameters):
         args, env = self.settings_to_cmd_args_env(self.connection.settings_dict, parameters)
-        if env:
-            env = {**os.environ, **env}
+        env = {**os.environ, **env} if env else None
```

Comment:
I don't get why you guys prefer small diffs over clean code, that naming and reassignment is the reason of the bug.
If the variable was named extra_env then I'm sure that would prevent that bug, and reassignment in general makes reading code harder.

Reply:
I think the variable names have nothing to do with this regression. Also it's much easier to fix a regression introduced in commit with a small diff, that's why we prefer small diffs for patches that will be backported.  

P.S. Note that we're not all "guys" so please use gender neutral greetings, https://heyguys.cc/